pkg/controller/bundle_sync_task.go

```diff
@@ -212,33 +212,48 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		readyCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceReady, Status: smith_v1.ConditionFalse}
 		errorCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceError, Status: smith_v1.ConditionFalse}
 
-		resInfo := st.processedResources[res.Name]
```

Comment:
This was causing NPE if the resource was not processed because something failed earlier e.g. the topological sort.